File: parking-front/src/app/models/recharge-transaction.ts
import { Transaction } from "./transaction";

export class RechargeTransaction implements Transaction{
    type: string;
    date: Date;
    amount: number;
    newBalance: number;
    operation: string;

    constructor(type:string, date: Date, amount: number, newBalance: number, operation: string){
        this.type = type;
        this.date = date;
        this.amount = amount;
        this.newBalance = newBalance;
        this.operation = operation;
    }

    public getAmountColor(): string {
        return '$green-600';
    }
}

File: parking-back/src/main/java/cespi/induccion/estacionamiento/models/RechargeTransaction.java
package cespi.induccion.estacionamiento.models;

import cespi.induccion.estacionamiento.DTO.RechargeTransactionDTO;
import jakarta.persistence.Entity;

@Entity
public class RechargeTransaction extends Transaction{
	
	
	public RechargeTransaction() {
		super();
	}
	
	public RechargeTransaction(double monto, double newBalance, String operation) {
		super(monto, newBalance, operation);
	}
	
	public RechargeTransactionDTO getDTO() {
		return new RechargeTransactionDTO(this.getDate(), this.getAmount(), this.getNewBalance(), this.getOperation());
	}

}
